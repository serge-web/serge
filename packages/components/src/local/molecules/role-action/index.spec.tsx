/* global it expect */
import React from 'react'
import renderer from 'react-test-renderer'
import RoleAction from './index'

describe('RoleAction component:', () => {
  it('renders correctly', () => {
    const editRole = (): void => window.alert('Edit role clicked')
    const removeRole = (): void => window.alert('Delete role clicked')
    const tree = renderer
      .create(
        <RoleAction role="CO" onEditRole={editRole} onRemoveRole={removeRole} />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
