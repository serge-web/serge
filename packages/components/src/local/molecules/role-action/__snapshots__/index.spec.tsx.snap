// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoleAction component: renders correctly 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root makeStyles-root-1 makeStyles-root-3 MuiButton-contained makeStyles-contained-2 makeStyles-contained-4"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <span
      className="button-inner"
    >
      <span>
        CO
      </span>
      <svg
        className="MuiSvgIcon-root role-action-icons"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
        />
        <title>
          Edit this role
        </title>
      </svg>
      <svg
        className="MuiSvgIcon-root role-action-icons"
        focusable="false"
        onClick={[Function]}
        role="img"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
        <title>
          Remove this role
        </title>
      </svg>
    </span>
  </span>
</button>
`;
